sql/core/src/main/scala/org/apache/spark/sql/api.scala

```diff
@@ -171,6 +171,27 @@ private[sql] trait DataFrameSpecificApi {
   def saveAsTable(tableName: String): Unit
 
   @Experimental
+  def saveAsTable(tableName: String, dataSourceName: String, options: Map[String, String]): Unit
```

Comment:
When we add the short hand version that only takes a table name, we need to make sure that we set the `EXTERNAL` property correctly so that `DROP TABLE` will operate as users expect.